File: leetcode2968/contribution.cpp
// 滑动窗口枚举右端点
// 计算窗口滑动时，各个元素的变换引起的差值
// 注意：窗口中位数的计算，根据原始窗口尺寸的奇偶性进行讨论
// class Solution {
// public:
//     int maxFrequencyScore(vector<int> &nums, long long k) {
//         sort(nums.begin(), nums.end());
//         int ans = 0, left = 0;
//         long long s = 0; // 窗口元素与窗口中位数的差之和
//         for (int right = 0; right < nums.size(); right++) {
//             s += nums[right] - nums[(left + right) / 2];
//             while (s > k) {
//                 s += nums[left] - nums[(left + right + 1) / 2];
//                 left++;
//             }
//             ans = max(ans, right - left + 1);
//         }
//         return ans;
//     }
// };

File: DataStructure/leetcode2968/contribution.go
// func maxFrequencyScore(nums []int, k int64) int {
//     n:=len(nums)
//     sort.Ints(nums)

//     prefix:=make([]int64,n+1)

//     for i:=0;i<n;i++{
//         prefix[i+1]=prefix[i]+int64(nums[i])
//     }

//     ans:=0
//     var s int64=0
//     left:=0
//     for i:=0;i<n;i++{
//         s+=int64(nums[i]-nums[(left+i)>>1])
//         for s>k{
//             s+=int64(nums[left] - nums[(left+i+1)>>1])
//             left++
//         }
//         ans=max(ans,i-left+1)
//     }

//     return ans
// }